src/hooks/usePostFetch.js

```diff
@@ -0,0 +1,36 @@
+import { useState, useEffect } from "react";
+
+export default function usePostFetch(url, numbersArray) {
+  const [response, setResponse] = useState(null);
+
+  useEffect(() => {
+    const encodeBoard = (board) =>
+      board.reduce(
+        (result, row, i) =>
+          result +
+          `%5B${encodeURIComponent(row)}%5D${
+            i === board.length - 1 ? "" : "%2C"
+          }`,
+        ""
+      );
+
+    const encodeParams = (params) =>
+      Object.keys(params)
+        .map((key) => key + "=" + `%5B${encodeBoard(params[key])}%5D`)
+        .join("&");
+    const data = {
+      board: numbersArray,
+    };
```

Comment:
This is super hard to read. First thought, that the `const data` was part of the `encodeParams` function. Would recommend putting an empty line in between.
```suggestion
    const encodeParams = (params) =>
      Object.keys(params)
        .map((key) => key + "=" + `%5B${encodeBoard(params[key])}%5D`)
        .join("&");
        
    const data = {
      board: numbersArray,
    };
```